### ISSUE TITLE
Visiting the following page completely stops browser on MacOS

### ISSUE THREAD

**Issue by butlermatt** [brave/browser-laptop#600]
Trying to view this article:
http://www.xaverian.ca/2014/10/30/mother-webbs-exposed-post-on-mother-webbs-facebook-page-brings-to-light-accounts-of-sexual-harassment/

Brave goes to a blank gray screen. Cannot close it, cannot quit. Need to use kill -KILL <pid> to end the session after which I can then close the window and restart.

Mac OS X 10.11.3
Brave: 0.7.12

**Comment by Codes4Fun** [comment]
I reproduced this on Window 7, except it goes black (navigation bar included).

I traced it down to this line of html:
```
<link rel="Shortcut Icon" href="http://www.xaverian.ca/wp-content/themes/xaverian/images/favicon.ico" type="image/x-icon">
```

**Comment by Codes4Fun** [comment]
The icon file itself seems to be bugged, it reports a width of 256x256 but the internal PNG is 16x16, if I take the icon into a hexeditor and change bytes 6 and 7 to 0x10 (16) the bug doesn't happen. Looking at other browsers, the other browsers don't use that icon, I'm trying to find which icons they do use and where it is located.

**Comment by Codes4Fun** [comment]
So the icon the other browsers use come from this html:
```
<link rel="icon" href="http://www.xaverian.ca/wp-content/uploads/2015/09/cropped-Circle-logo-32x32.png" sizes="32x32">
```
Chrome uses the first icon it finds in the html and if it fails tries to use the next icon.
Firefox uses the last icon and if it fails displays no icon.
So it looks like Brave simply needs to detect the bad icon and respond accordingly.

**Comment by Codes4Fun** [comment]
This issue seems to be with vibrant.js drag/dropping the icon on their site http://jariz.github.io/vibrant.js/ makes chrome display the sad face, but firefox throws an exception cmap.vboxes is undefined.

In Brave if I pass an empty string instead of getFavicon the issue doesn't happen, the icon doesn't show up on the tab.

**Comment by bbondy** [comment]
Nice find! You guys made it much easier to fix, I'll take the fix.

**Comment by bbondy** [comment]
So I found that it hangs as mentioned before in Vibrant, but more specifically its' not vibrant's fault, it's libchromiumcontent's context.drawImage when passed an invalid image 
`context.drawImage(b, 0, 0, 256, 256)`

I tried work around like resizing the image first but that didn't work.  The image onload actually does succeed as well and libchromiumcontent thinks the width and height are 256.

We could perhaps always pick a png before an ico when multiple are available.  That format wouldn't have this problem.  But it wouldn't solve the root issue which would happen when no png was available and you have a bad icon like this.
